lib/valueflow.cpp

```diff
@@ -4011,9 +4012,10 @@ struct LifetimeStore {
             const Variable *var = getLifetimeVariable(tok2, er);
             // TODO: the inserted data is never used
             er.insert(er.end(), errorPath.begin(), errorPath.end());
-            if (!var)
+            if (!var || !tok)
```

Comment:
The check for `tok` should be moved outside the loop.

Reply:
Yeah - that's even better. I just did so not to introduce `declEndToken()` call where there was none before.

While we are at this code - please take a look at the TODO as well.